### ISSUE TITLE
"File In Files progress…" is always shown

### ISSUE THREAD

**Issue by vlakoff** [notepad-plus-plus/notepad-plus-plus#293]
After upgrading to 6.7.9.1 (from 6.7.8.2 which hadn't this issue), the "File In Files progress…" progress dialog is always shown, *even if I switch to another application*. The dialog is even visible when playing a movie fullscreen. This is quite surprising, and probably not intended.

**Comment by vlakoff** [comment]
Also, when the search is done, the Notepad++ window takes focus. I was typing something in another application, and suddenly got myself typing into Notepad++.

**Comment by pnedev** [comment]
Hi,

I have modified the Find In Files progress that way.
@milipili, that is not a bug, at least I'm not considering it to be one.
@vlakoff, the progress is made to be always on top because otherwise, due to the Find In Files implementation if you click on the main N++ window the progress will be hidden and this is highly undesirable. I seriously doubt that somebody would run Find In Files operation and start watching a movie at the same time. Besides you can always move the progress aside.
Anyway, unless Find In Files implementation in N++ is totally changed this is the lesser evil IMO.

About the focus, I thought that it would be helpful to the user to show and focus on N++ window on Find In Files completion as a notification and automatic displaying of the results. Otherwise one would constantly need to manually check if the search is ready or not. Perhaps it is a matter of preference.

Thanks for the feedback!

**Comment by vlakoff** [comment]
Thank you for these detailed explanations :)

I largely disagree… I was going to elaborate, but just a few points:
* I sometimes launch searches which are lenghty, and I shouldn't be the only one doing so. Now during this time I have a residual window on top of my porn. It impacts all applications, not just movies. Moving the window to a screen edge is not a serious solution…
* About the focus, I just *hate* when applications take focus. I've heard enough ranting in my life to attest I'm not the only one feeling the same!

So basically, my opinions are the opposite of yours ;)

Could other users provide their feedback to settle on this?

**Comment by MAPJe71** [comment]
Instead of taking focus you could flash the taskbar entry and keep keep flashing it with an interval until the user changes focus to the application requesting attention.

**Comment by pnedev** [comment]
@vlakoff, as I said it is a matter of preference and I respect your personal opinion, thanks for sharing it :)
@MAPJe71, your proposal sounds great, thanks!

I'm not sure I'll have time soon to work on this though...

**Comment by RiccardoB** [comment]
I second @vlakoff request.
In my specific case I have to work with projects made by third parties and I have to search for strings on big folders where the search takes also more than 10 minutes to complete and during the search I work on other part of project (like design) and the animated progress bar on top is annoying.
If possible I suggest to add a preference to switch this behavior.

**Comment by MekalaChitti** [comment]
I have the same issue. I have to work on multiple applications and this progress bar is on top and makes seeing the screen difficult. An option to choose the preference would be great.

**Comment by donho** [comment]
It may be an intentional behaviour but it's surely a unexpected result for the most of users (including me). I'll check myself if @pnedev is not available.
